src/libs/PolicyUtils.ts

```diff
@@ -0,0 +1,217 @@
+import Str from 'expensify-common/lib/str';
+import CONST from '../CONST';
+import ONYXKEYS from '../ONYXKEYS';
+import * as OnyxTypes from '../types/onyx';
+
+type PolicyMemberList = Record<string, OnyxTypes.PolicyMember>;
+type PolicyMembersCollection = Record<string, PolicyMemberList>;
+type MemberEmailsToAccountIDs = Record<string, string>;
+type PersonalDetailsList = Record<string, OnyxTypes.PersonalDetails>;
+type UnitRate = {rate: number};
+
+/**
+ * Filter out the active policies, which will exclude policies with pending deletion
+ * These are policies that we can use to create reports with in NewDot.
+ */
+function getActivePolicies(policies: OnyxTypes.Policy[]): OnyxTypes.Policy[] {
+    return (policies ?? []).filter(
+        (policy) => policy && (policy.isPolicyExpenseChatEnabled || policy.areChatRoomsEnabled) && policy.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE,
+    );
+}
+
+/**
+ * Checks if we have any errors stored within the POLICY_MEMBERS. Determines whether we should show a red brick road error or not.
+ * Data structure: {accountID: {role:'user', errors: []}, accountID2: {role:'admin', errors: [{1231312313: 'Unable to do X'}]}, ...}
+ */
+function hasPolicyMemberError(policyMembers: PolicyMemberList): boolean {
```

Comment:
```suggestion
function hasPolicyMemberError(policyMembers: OnyxEntry<OnyxTypes.PolicyMembers>): boolean {
```

I noticed that the types for `ONYXKEYS.COLLECTION.DEPRECATED_POLICY_MEMBER_LIST` are wrong in `ONYXKEYS.ts`. I would suggest adding a `PolicyMembers` type to `src/types/onyx/PolicyMember.ts` file, exporting and using in `ONYXKEYS.ts`. Here, you can use `PolicyMembers` type directly.

```ts
// src/types/onyx/PolicyMember.ts
import * as OnyxCommon from './OnyxCommon';

type PolicyMember = {
    /** Role of the user in the policy */
    role?: string;

    /**
     * Errors from api calls on the specific user
     * {<timestamp>: 'error message', <timestamp2>: 'error message 2'}
     */
    errors?: OnyxCommon.Errors;

    /** Is this action pending? */
    pendingAction?: OnyxCommon.PendingAction;
};

type PolicyMembers = Record<string, PolicyMember>;

export default PolicyMember;
export type {PolicyMembers};

// src/ONYXKEYS.ts
[ONYXKEYS.COLLECTION.DEPRECATED_POLICY_MEMBER_LIST]: OnyxTypes.PolicyMembers;
```